node/api.go

```diff
@@ -58,14 +59,30 @@ func (api *PrivateAdminAPI) AddPeer(url string) (bool, error) {
 }
 
 // StartRPC starts the HTTP RPC API server.
-func (api *PrivateAdminAPI) StartRPC(host string, port int, cors string, apis string) (bool, error) {
+func (api *PrivateAdminAPI) StartRPC(host *string, port *rpc.HexNumber, cors *string, apis *string) (bool, error) {
```

Comment:
I'm not really sure all these pointers are warranted for. We could do all these checks for the default values too, not sure why adding pointer complexity in makes a difference.

* `cors`: pointer conversion just needs extra initialization to the empty string, which would have been the default any way
* `apis`: since it doesn't make sense to start an endpoint with no APIs, we can default the empty string to `rpc.DefaultHTTPApis`, no need to go through pointers

For the `host` and `port` this new code indeed is a bit more flexible as it allows differentiating between **default** values and valid zero values (i.e. all interfaces and the random port).